easybuild/tools/run.py

```diff
@@ -97,6 +97,32 @@ def cache_aware_func(cmd, *args, **kwargs):
     return cache_aware_func
 
 
+def get_output_from_process(proc, read_size=None, asynchronous=False):
+    """
+    Get output from running process (that was opened with subprocess.Popen).
+
+    :param proc: process to get output from
+    :param read_size: number of bytes of output to read (if None: read all output)
+    :param asynchronous: get output asynchronously
+    """
+
+    if asynchronous:
+        output = recv_some(proc)
+    elif read_size:
+        output = proc.stdout.read(read_size)
+    else:
+        output = proc.stdout.read()
```

Comment:
Do we need some sort of check in the case the function is called with `read_size=1234, asynchronous=True`?

Reply:
That's not a valid use case currently... `asynchronous=True` is used by `run_cmd_qa` to get intermediate output, `read_size` is used by `run_cmd` to get output in a buffered way

Reply:
I added a dedicated test for `get_output_from_process` (cfr. a37a08408), which shows that `read_size` is effectively ignored when using `asynchronous=True`...